### PR TITLE
implement demotest

### PR DESCRIPTION
**Pull request by rfomin**
I haven't added all the demos yet.

It works, but maybe submodule or other repository is better?

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
I like this idea!

File: src/i_system.c [hunk 1]
```diff
@@ -150,7 +155,11 @@ void I_SafeExit(int rc)
     }
   }
 
-  exit(rc);
+  #ifdef WIN_LAUNCHER
```
Comment (fabiangreffrath) [inline src/i_system.c:158]: What is this for?

Comment (rfomin) [reply src/i_system.c:158]: `exit()` does not work on Windows when called from a DLL. On Windows have the woof.exe and woof.com launcher programs, and all the code is in woof.dll.

File: demotest/demotest [hunk 2]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (rfomin) [reply demotest/demotest:1]: But quickcheck has no copyrights. This code is very trivial 😄

File: src/i_system.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/i_system.c:158]: I see, thanks!

File: demotest/demotest (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply demotest/demotest:1]: I guess it falls under the projects' collective copyright and GPL-2.0+ then. 😉

File: demotest/demotest [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    sys.stdout.write(line)
+
+    if error:
+        sys.exit(1)
+    else:
+        print("Success.")
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description='Execute demos for Doom port in a batch')
+    parser.add_argument('--jobs', dest='jobs', type=int, help='Set the number of jobs')
```
Comment (fabiangreffrath) [inline demotest/demotest:88]: Should these get default values?

Comment (fabiangreffrath) [reply demotest/demotest:88]: E.g. `1` for `--jobs` and `doom` for `--port`?

Comment (rfomin) [reply demotest/demotest]: Fixed in https://github.com/fabiangreffrath/woof/pull/1260/commits/72ec89d72488c29a3e3dc194caaa7295625789e6

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**